src/main/java/com/microsoft/azure/servicebus/jms/ServiceBusJmsConnectionFactory.java

```diff
@@ -83,21 +101,49 @@ public ServiceBusJmsConnectionFactory(ConnectionStringBuilder connectionStringBu
      */
     public ServiceBusJmsConnectionFactory(String sasKeyName, String sasKey, String host, ServiceBusJmsConnectionFactorySettings settings) {
         this.settings = settings;
-        this.initialize(sasKeyName, sasKey, host, settings);
+        this.setKeyValue(sasKey);
+        this.keyName = sasKeyName;
+        this.initialize(host, settings);
     }
     
-    private void initialize(String sasKeyName, String sasKey, String host, ServiceBusJmsConnectionFactorySettings settings) {
-        if (sasKeyName == null || sasKeyName == null || host == null) {
-            throw new IllegalArgumentException("SAS Key, SAS KeyName and the host cannot be null for a ServiceBus connection factory.");
-        }
-        
+    /**
+     * Create a ServiceBusJmsConnectionFactory using shared access key and host name.
+     * @param tenantId Tenent id 
+     * @param clientId Client id.
+     * @param clientSecret Client secret
+     * @param host The host name of the ServiceBus namespace. Example: your-namespace-name.servicebus.windows.net
+     * @param settings The options used for this ConnectionFactory. Null can be used as default.
+     */
+    public ServiceBusJmsConnectionFactory(String tenantId, String clientId, String clientSecret, String host, ServiceBusJmsConnectionFactorySettings settings) {
+        this.settings = settings;
+        this.setIsToken(true);
```

Comment:
>true [](http://example.com/codeflow?start=24&length=4)

nit: Can we classify the variable along with the true flag, I mean something like isToken:true